extensions/amp-analytics/0.1/events.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Observable} from '../../../src/observable';
+import {getDataParamsFromAttributes} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+const VARIABLE_DATA_ATTRIBUTE_KEY = /^vars(.+)/;
+
+
+/**
+ * The analytics event.
+ */
+export class AnalyticsEvent {
+  /**
+   * @param {!Element} target The most relevant target element.
+   * @param {string} type The type of event.
+   * @param {!Object<string, string>=} opt_vars A map of vars and their values.
+   */
+  constructor(target, type, opt_vars) {
+    /** @const */
+    this.target = target;
+    /** @const */
+    this.type = type;
+    /** @const */
+    this.vars = opt_vars || Object.create(null);
+  }
+}
+
+
+/**
+ * The base class for all trackers. A tracker tracks all events of the same
+ * type for a single analytics root.
+ *
+ * @implements {../../../src/service.Disposable}
+ * @abstract
+ * @visibleForTesting
+ */
+export class EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    /** @const */
+    this.root = root;
+  }
+
+  /** @override @abstract */
+  dispose() {}
+
+  /**
+   * @param {!Element} unusedContext
+   * @param {string} unusedEventType
+   * @param {!JSONType} unusedConfig
+   * @param {function(!AnalyticsEvent)} unusedListener
+   * @return {!UnlistenDef}
+   * @abstract
+   */
+  add(unusedContext, unusedEventType, unusedConfig, unusedListener) {}
+}
+
+
+/**
+ * Tracks custom events.
+ */
+export class CustomEventTracker extends EventTracker {
```

Comment:
It's created/used here: https://github.com/ampproject/amphtml/pull/7172/files#diff-f979d4b9254b24d9616a865981a47cd6R553

Notice, since some time ago, GitHub started to arbitrarily collapse some diffs and just puts "Load diff" instead. Drives me nuts...